inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp

```diff
@@ -113,12 +116,80 @@ class SharedObjectLoader::Impl {
     }
 #endif
 
-public:
+    static const char  kPathSeparator = '\\';
+
+    static const char* FindLastPathSeparator(LPCSTR path) {
+        const char* const last_sep = strchr(path, kPathSeparator);
+        return last_sep;
+    }
+
+    std::basic_string<CHAR> GetDirname(LPCSTR path) {
+        auto pos = FindLastPathSeparator(path);
+        if (pos == nullptr) {
+            return path;
+        }
+        std::basic_string<CHAR> original(path);
+        original[pos - path] = 0;
+        return original;
+    }
+
 #ifdef ENABLE_UNICODE_PATH_SUPPORT
+    static const wchar_t* FindLastPathSeparator(LPCWSTR path) {
+        const wchar_t* const last_sep = wcsrchr(path, kPathSeparator);
+        return last_sep;
+    }
+
+    std::basic_string<WCHAR> GetDirname(LPCWSTR path) {
+        auto pos = FindLastPathSeparator(path);
+        if (pos == nullptr) {
+            return path;
+        }
+        std::basic_string<WCHAR> original(path);
+        original[pos - path] = 0;
+        return original;
+    }
+
+    std::basic_string<WCHAR> IncludePluginDirectory(LPCWSTR path) {
+        DWORD nBufferLength = GetDllDirectoryW(0, nullptr);
```

Comment:
we need to use `GetDllDirectoryA_Fnc` and so on. These functions are not available on UWP platforms, so we need dynamically decide whether can use `kernel32.dll` or not

Reply:
modified, but need to define the right condition, for now, I changed it to !WINAPI_PARTITION_SYSTEM